Improve KMS documentation (#493)
digital-asset/docs.daml.com: docs/2.8.0/docs/canton/usermanual/kms/external_key_storage.rst

..
   Copyright (c) 2023 Digital Asset (Switzerland) GmbH and/or its affiliates.
..
   Proprietary code. All rights reserved.

.. _external_key_storage:

> Configure External Key Storage and Usage with a Key Management Service (KMS)

Reviewer: I think `Configure External Key Storage` is sufficient, WDYT? Seems implied that we're going to use them

```suggestion
Configure External Key Storage with a Key Management Service (KMS)
```

============================================================================

.. enterprise-only::

> Canton can use a KMS to `generate and store the private keys itself`.
> We directly use a KMS to generate and store Canton's private keys
> and then use its API to securely sign an decrypt messages. A Canton node still stores
> the corresponding public keys in its stores so that it can verify signatures and
> encrypt messages without having to rely on the KMS.

Reviewer: It's not clear who `itself` refers to here. Is it Canton or the KMS? Maybe a better phrasing would be

```suggestion
Canton can delegate all asymmetric decryption and signing operations to a KMS. In that mode, private keys
are stored on the KMS, not in Canton's own storage nor in memory. Decryption and signing operations each
require an API call to the KMS. A Canton node still stores
the corresponding public keys in its stores so that it can verify signatures and
encrypt messages without having to rely on the KMS.
```

> If you wish to know more you can read section: :ref:`Externalize Private Keys With a Key Management Service <kms_external_architecture>`.

Reviewer: ```suggestion
See :ref:`Externalize Private Keys With a Key Management Service <kms_external_architecture>` for more details.
```

> Currently Canton supports external key storage and usage with an `Amazon Web Services (AWS) KMS

Reviewer: ```suggestion
Currently Canton supports external key storage with `Amazon Web Services (AWS) KMS
```

<https://aws.amazon.com/kms/>`_ or

> a `Google Cloud Provider (GCP) KMS <https://cloud.google.com/security-key-management?hl=en>`_.

Reviewer: ```suggestion
`Google Cloud Provider (GCP) KMS <https://cloud.google.com/security-key-management?hl=en>`_.
```
